controllers/precache.go

```diff
@@ -38,7 +54,7 @@ func (r *ClusterGroupUpgradeReconciler) reconcilePrecaching(
 			return nil
 		}
 		// Precaching is required and not marked as done
-		return r.precachingFsm(ctx, clusterGroupUpgrade)
+		return r.precachingFsm(ctx, clusterGroupUpgrade.Status.Precaching.Clusters, clusterGroupUpgrade)
```

Comment:
Why are we passing both `clusterGroupUpgrade` and its inner structure?

Reply:
Fixed as suggested with https://github.com/openshift-kni/cluster-group-upgrades-operator/commit/8af5d7ad847ce4a50b06e95e0923bcb71563855c